api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java

```diff
@@ -12,24 +12,36 @@
 @Service
 @Slf4j
 @RequiredArgsConstructor
-@SuppressWarnings("PMD.DataflowAnomalyAnalysis") // PMD does not like reassignment of `eligibilityStatus`
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis", "PMD.AvoidCatchingGenericException"}) // PMD does not like reassignment of `eligibilityStatus`
 public class ClaimService {
 
     private final ClaimantRepository claimantRepository;
     private final EligibilityClient client;
 
-    public void createClaim(Claim claim) {
+    public Claimant createClaim(Claim claim) {
         Claimant claimant = claim.getClaimant();
         EligibilityStatus eligibilityStatus = EligibilityStatus.ERROR;
 
         try {
-            EligibilityResponse eligibilityResponse = client.checkEligibility(claimant);
-            eligibilityStatus = eligibilityResponse.getEligibilityStatus();
-            claimant.setHouseholdIdentifier(eligibilityResponse.getHouseholdIdentifier());
-        } finally {
-            claimant.setEligibilityStatus(eligibilityStatus);
-            claimantRepository.save(claimant);
-            log.info("Saved new claimant: {} with status {}", claimant.getId(), claimant.getEligibilityStatus());
+            if (claimantRepository.claimantExists(claimant.getNino())) {
+                eligibilityStatus = EligibilityStatus.DUPLICATE;
+            } else {
+                EligibilityResponse eligibilityResponse = client.checkEligibility(claimant);
+                eligibilityStatus = eligibilityResponse.getEligibilityStatus();
+                claimant.setHouseholdIdentifier(eligibilityResponse.getHouseholdIdentifier());
+            }
+
+            saveClaimant(claimant, eligibilityStatus);
+            return claimant;
+        } catch (Exception e) {
```

Comment:
looks like you don't need this catch block, just put 
```
            saveClaimant(claimant, eligibilityStatus);
            return claimant;
```
in a `finally` block

Reply:
actually, not the `return claimant` - just saveClaimant

Reply:
The thing we like about this way is that you can then remove the `eligibilityStatus` field being outside the try block and default it in the catch block. So its swapping one PMD warning for another.

Reply:
There are pros and cons to both from the discussions Gethyn and I are currently having.